src/app/components/Card/Card.tsx

```diff
@@ -30,7 +30,7 @@ export default function Card({
       <article className={styles.card__article}>
         <section className={styles.card__header}>
           <div className={styles.card__title}>
-            <h2>{groupTitle}</h2>
+            <h3>{groupTitle}</h3>
```

Comment:
is the groupTitle semantically a h3? If not, and you changed it to make the font smaller, consider doing so in css instead.